activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -215,7 +226,9 @@ def column_exists?(table_name, column_name, type = nil, options = {})
       def create_table(table_name, options = {})
         td = create_table_definition table_name, options[:temporary], options[:options], options[:as]
 
-        if options[:id] != false && !options[:as]
+        if options[:id] == false
+          td.primary_keys options[:primary_key] if options[:primary_key].is_a?(Array)
```

Comment:
Think it's odd that we'd have to set `id: false` for `primary_keys: […]` to take effect.